### PR TITLE
feat(prejoin_page): Rework prejoin page

### PR DESCRIPTION
**Pull request by vp8x8**
* Add a checkbox for skipping the prejoin page on next use. (This is hidden for
now, until we also have a settings entry for it).
* Rework 'Join by Phone' buttons and add new overlay.
* Update the device status accordingly if there were errors while adding
devices.
* The input is filled with the display name if there was one previously used.
* Join the meeting on 'Enter' press.

Screenshot 2020-05-07 at 10 23 59
Screenshot 2020-05-07 at 10 24 26

### CODE REVIEW

File: react/features/base/settings/reducer.js [hunk 1]
```diff
@@ -125,6 +126,7 @@ function _initSettings(featureState) {
     // jibri, and remove the old settings.js values.
     const savedDisplayName = jitsiLocalStorage.getItem('displayname');
     const savedEmail = jitsiLocalStorage.getItem('email');
+    const userSelectedSkipPrejoin = jitsiLocalStorage.getItem('userSelectedSkipPrejoin');
```
Comment (saghul) [inline react/features/base/settings/reducer.js:129]: No need for this. Jibri is now ready to override the value in config.js

File: react/features/prejoin/components/preview/Preview.js [hunk 2]
```diff
@@ -66,7 +67,6 @@ function Preview(props: Props) {
  */
 function mapStateToProps(state) {
     return {
-        name: getPrejoinName(state),
```
Comment (saghul) [inline react/features/prejoin/components/preview/Preview.js:69]: remove this from props? or is it now passed explicitly?

File: react/features/prejoin/middleware.js [hunk 3]
```diff
@@ -70,26 +76,16 @@ MiddlewareRegistry.register(store => next => async action => {
         store.dispatch(setPrejoinVideoMuted(Boolean(action.muted)));
         break;
     }
+
+    case SET_PREJOIN_NAME: {
```
Comment (saghul) [inline react/features/prejoin/middleware.js:80]: Why are we disspatching this action now? Maybe we can just dispatch updateSettings?

File: react/features/base/settings/reducer.js (hunk 1 above)
Comment (vp8x8) [reply react/features/base/settings/reducer.js]: Removed it.

File: react/features/prejoin/middleware.js (hunk 3 above)
Comment (vp8x8) [reply react/features/prejoin/middleware.js]: I have removed the redundant action.

File: react/features/prejoin/components/preview/Preview.js (hunk 2 above)
Comment (vp8x8) [reply react/features/prejoin/components/preview/Preview.js:69]: Yes, this is now passed explicitly.

File: react/features/prejoin/middleware.js [hunk 4]
```diff
@@ -51,10 +50,16 @@ MiddlewareRegistry.register(store => next => async action => {
     }
 
     case PREJOIN_START_CONFERENCE: {
-        const { dispatch, getState } = store;
+        const { getState, dispatch } = store;
+        const state = getState();
+        const { userSelectedSkipPrejoin } = state['features/prejoin'];
+
+        userSelectedSkipPrejoin && dispatch(updateSettings({
```
[the remark this answers is not in the document]
Comment (vp8x8) [reply react/features/prejoin/middleware.js:57]: If we would have it directly linked to the actual store value, when the user checks it, will automatically join the meeting. But we don't want this. We want the user to join when it clicks the 'Join' button or presses 'Enter' and keep this setting only for future uses.

**Pull request merged**